tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeVstsAgent.cs

```diff
@@ -0,0 +1,155 @@
+    using System.Linq;
+    using DevOpsLib.VstsModels;
+
+    public class IoTEdgeVstsAgent : IEquatable<IoTEdgeVstsAgent>, ICloneable
+    {
+        readonly int id;
+        readonly string name;
+        readonly string version;
+        readonly VstsAgentStatus status;
+        readonly bool enabled;
+        readonly HashSet<AgentCapability> systemCapabilities;
+        readonly HashSet<AgentCapability> userCapabilities;
+
+        public IoTEdgeVstsAgent(int id, string name, string version, VstsAgentStatus status, bool enabled, HashSet<AgentCapability> systemCapabilities, HashSet<AgentCapability> userCapabilities)
+        {
+            ValidationUtil.ThrowIfNonPositive(id, nameof(id));
+            ValidationUtil.ThrowIfNullOrWhiteSpace(name, nameof(name));
+            ValidationUtil.ThrowIfNullOrWhiteSpace(version, nameof(version));
+            ValidationUtil.ThrowIfNulOrEmptySet(systemCapabilities, nameof(systemCapabilities));
+
+            this.id = id;
+            this.name = name;
+            this.version = version;
+            this.status = status;
+            this.enabled = enabled;
+            this.systemCapabilities = systemCapabilities;
+            this.userCapabilities = userCapabilities ?? new HashSet<AgentCapability>();
+        }
+
+        public int Id => this.id;
+
+        public string Name => this.name;
+
+        public string Version => this.version;
+
+        public bool Enabled => this.enabled;
+
+        public VstsAgentStatus Status => this.status;
+
+        public ImmutableHashSet<AgentCapability> SystemCapabilities => this.systemCapabilities.ToImmutableHashSet();
+
+        public ImmutableHashSet<AgentCapability> UserCapabilities => this.userCapabilities.ToImmutableHashSet();
+
+        public bool IsAvailable => this.enabled && this.status == VstsAgentStatus.Online;
+
+        public static IoTEdgeVstsAgent Create(VstsAgent vstsAgent)
+        {
+            HashSet<AgentCapability> systemCapabilities = vstsAgent.SystemCapabilities?.Select(c => new AgentCapability(c.Key, c.Value)).ToHashSet();
+            HashSet<AgentCapability> userCapabilities = vstsAgent.UserCapabilities?.Select(c => new AgentCapability(c.Key, c.Value)).ToHashSet();
+
+            return new IoTEdgeVstsAgent(
+                vstsAgent.Id,
+                vstsAgent.Name,
+                vstsAgent.Version,
+                vstsAgent.Status,
+                vstsAgent.Enabled,
+                systemCapabilities,
+                userCapabilities);
+        }
+
+        public bool Match(IEnumerable<AgentCapability> capabilities)
+        {
+            var allCapabilities = this.systemCapabilities.Union(this.userCapabilities);
+            return capabilities.All(allCapabilities.Contains);
+        }
+
+        public string GetAgentGroup()
+        {
+            return this.userCapabilities.Where(c => c.Name.Equals("agent-group")).Select(c => c.Value).FirstOrDefault() ?? string.Empty;
```

Comment:
Per the pervious comment about const class, I think `agent-group` would be a good candidate to go into the const class.